### PR TITLE
Grap 11

### PR DESCRIPTION
**Pull request by ppmarkek**
- Localization (at least 2 languages). You should be able to change the language by clicking on the toggler/select in the header. Localization should be implemented using Context API. No 3rd party / open-source solutions are allowed
- Implement language switch functionality

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
+import { useState } from 'react';
 import Header from './components/templates/Header/Header';
+import { LenguageContext } from './components/Context/LenguageContext';
 
 function App() {
+  const [language, setLanguage] = useState('');
```
Comment (1LisaW) [inline src/App.tsx]: please add some default value, for example "en"

File: src/components/Context/LenguageContext.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Dispatch, SetStateAction, createContext } from 'react';
+
+interface LenguageContextType {
+  language: string;
```
Comment (1LisaW) [inline src/components/Context/LenguageContext.ts]: enum Langs

File: src/components/Context/LenguageContext.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  setLanguage: Dispatch<SetStateAction<string>>;
+}
+
+export const initialState = {
+  language: '',
```
Comment (1LisaW) [inline src/components/Context/LenguageContext.ts]: defaul val != ''

File: src/components/Context/tests/LenguageContext.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import userEvent from '@testing-library/user-event';
+import { LenguageContext, initialState } from '../LenguageContext';
+
+const TestComponent = () => {
+  const { language, setLanguage } = useContext(LenguageContext);
+
+  return (
+    <div>
+      <span data-testid="language-value">{language}</span>
+      <button onClick={() => setLanguage('English')}>Set to English</button>
```
Comment (1LisaW) [inline src/components/Context/tests/LenguageContext.spec.tsx]: ISO code of lang (the same will be with json names)"en" https://www.loc.gov/standards/iso639-2/php/code_list.php

File: src/components/Context/tests/LenguageContext.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    </div>
+  );
+};
+
+interface TestWrapperProps {
+  children: ReactNode;
+}
+
+const TestWrapper: React.FC<TestWrapperProps> = ({ children }) => {
+  const [language, setLang] = useState('');
```
Comment (1LisaW) [inline src/components/Context/tests/LenguageContext.spec.tsx]: "en"

File: src/components/templates/Header/Header.tsx [hunk 6]
```diff
@@ -1,15 +1,27 @@
...
+  InputLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+} from '@mui/material';
+import { LenguageContext } from '../../Context/LenguageContext';
 
 const Header = () => {
   const [isSticky, setIsSticky] = useState(false);
+  const { language, setLanguage } = useContext(LenguageContext);
```
Comment (1LisaW) [inline src/components/templates/Header/Header.tsx]: LanguageContext - typo

**Review by 1LisaW — APPROVED** [review]

**Review by svyatoslavkk — APPROVED** [review]

**Pull request merged**